src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4846,6 +4846,1176 @@ .maxstack  1
+
+    [UnscopedRef]
+    public Span<float> Self
+    {  readonly get => throw null; set {}}
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics(
+                // (15,9): error CS1650: Fields of static readonly field 'Program.ReadOnlyVec' cannot be assigned to (except in a static constructor or a variable initializer)
+                //         ReadOnlyVec.Self = default;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyStatic2, "ReadOnlyVec.Self").WithArguments("Program.ReadOnlyVec").WithLocation(15, 9)
+                );
+        }
+
+        [Fact]
+        [WorkItem(64776, "https://github.com/dotnet/roslyn/issues/64776")]
+        public void DefensiveCopy_10()
+        {
+            var source =
+@"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+internal class Program
+{
+    private static readonly Vec4 ReadOnlyVec = new Vec4(1, 2, 3, 4);
+
+    static void Main()
+    {
+    }
+
+    private static void Test1()
+    {
+        ReadOnlyVec.Self = default;
+    }
+}
+
+public struct Vec4
+{
+    public float X, Y, Z, W;
+    public Vec4(float x, float y, float z, float w) => (X, Y, Z, W) = (x, y, z, w);
+
+    [UnscopedRef]
+    public Span<float> Self
+    {  get => throw null; readonly set {}}
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            CompileAndVerify(comp, verify: Verification.Skipped).VerifyDiagnostics().VerifyIL("Program.Test1",
+@"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  .locals init (System.Span<float> V_0)
+  IL_0000:  ldsflda    ""Vec4 Program.ReadOnlyVec""
+  IL_0005:  ldloca.s   V_0
+  IL_0007:  initobj    ""System.Span<float>""
+  IL_000d:  ldloc.0
+  IL_000e:  call       ""readonly void Vec4.Self.set""
+  IL_0013:  ret
+}
+");
+        }
+
+        [Fact]
+        [WorkItem(64776, "https://github.com/dotnet/roslyn/issues/64776")]
+        public void DefensiveCopy_11()
+        {
+            var source =
+@"
+
+using System.Diagnostics.CodeAnalysis;
+
+public struct Vec4
+{
+    public float X, Y, Z, W;
+    public Vec4(float x, float y, float z, float w) => (X, Y, Z, W) = (x, y, z, w);
+
+    [UnscopedRef]
+    public ref Vec4 Self2() => ref this;
+
+    [UnscopedRef]
+    readonly public ref Vec4 Test3()
+    {
+        ref Vec4 xyzw3 = ref this.Self2();
+        return ref xyzw3;
+    }
+
+    [UnscopedRef]
+    readonly public ref Vec4 Test4()
+    {
+        var r = this;
+        ref Vec4 xyzw4 = ref r.Self2();
+        return ref xyzw4;
+    }
+}";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyEmitDiagnostics(
+                // (16,30): warning CS8656: Call to non-readonly member 'Vec4.Self2()' from a 'readonly' member results in an implicit copy of 'this'.
+                //         ref Vec4 xyzw3 = ref this.Self2();
```

Comment:
Trying to reconcile this behavior with some behavior from earlier tests in `DefensiveCopy_02` 

```csharp
static ref Wrap m1(in Wrap i)
{
    ref Wrap r1 = ref i.Self; // defensive copy
    return ref r1; // ref to the local copy
}
```

This seems similar in that its a `ref` call to a member that causes a defensive copy to occur. In the case of `DefensiveCopy_02` we allowed the `ref` call without error but error'd when the local was attempted to be returned. In this case though we're erroring on both the defensive copy and the return. 

Is there something subtly different about this call site that I'm missing here?

Reply:
> Is there something subtly different about this call site that I'm missing here? 

If I am understanding the question correctly, in this case we are copying ```this```. Also, at the call site we are not reporting an error, it is a warning. And it is not about memory safety, but about accessing non-readonly member from a readonly one. This is yet another reason to perform a defensive copy and two different features report diagnostics because of that.

Reply:
Doh, I was reading that as an error, not a warning. This all makes sense now that I've read the actual words 😄